src/components/structures/FilePanel.js

```diff
@@ -40,42 +41,73 @@ const FilePanel = createReactClass({
         };
     },
 
-    componentDidMount: function() {
-        this.updateTimelineSet(this.props.roomId);
+    async componentDidMount() {
+        await this.updateTimelineSet(this.props.roomId);
     },
 
-    updateTimelineSet: function(roomId) {
+    async fetchFileEventsServer(room) {
         const client = MatrixClientPeg.get();
+
+        const filter = new Matrix.Filter(client.credentials.userId);
+        filter.setDefinition(
+            {
+                "room": {
+                    "timeline": {
+                        "contains_url": true,
+                        "types": [
+                            "m.room.message",
+                        ],
+                    },
+                },
+            },
+        );
+
+        // FIXME: we shouldn't be doing this every time we change room - see comment above.
+        // TODO: Remove this stale comment? Which comment above?
```

Comment:
hmm?

Reply:
The TODO is from me, the FIXME was there before. 🤷

Reply:
I guess just rip both out